trin-state/src/storage.rs

```diff
@@ -124,16 +128,22 @@ impl StateStorage {
         let trie_node = TrieNode {
             node: last_trie_node.clone(),
         };
-        self.store
+        match self
```

Comment:
I think this can be simplified to:
`self.store.insert(...).and(Ok(vec![]))`

Optionally and in addition, you can let these functions return dropped content and handle it once in a `put` function above.